ow_faults_detection/src/FaultDetector.cpp

```diff
@@ -133,31 +101,55 @@ void FaultDetector::jointStatesFlagCb(const ow_faults_detection::JointStatesFlag
     }
   }
 
-  bool armFault = false;
+  // check for arm faults
+  bool isFault = false;
   auto armList = {J_SHOU_YAW, J_SHOU_PITCH, J_PROX_PITCH, 
                   J_DIST_PITCH, J_HAND_YAW, J_SCOOP_YAW};
-  //ant faults
-  m_pan_fault = isFlagSet( J_ANT_PAN, msg->flags);
-  m_tilt_fault = isFlagSet( J_ANT_TILT, msg->flags);
-  antPublishFaultMessages();
-
-  //arm faults
   for (auto& name : armList) {
-    armFault = armFault || isFlagSet( name, msg->flags);
+    isFault = isFault || isFlagSet( name, msg->flags);
   }
-
-  owl_msgs::ArmFaultsStatus arm_faults_msg;
-  if (armFault){
+  if (isFault) {
     m_system_faults_flags |= SystemFaultsStatus::ARM_EXECUTION_ERROR;
-    arm_faults_msg.value |= ArmFaultsStatus::HARDWARE;
+    m_arm_faults_flags |= ArmFaultsStatus::HARDWARE;
   } else {
     m_system_faults_flags &= ~SystemFaultsStatus::ARM_EXECUTION_ERROR;
-    arm_faults_msg.value &= ~ArmFaultsStatus::HARDWARE;
+    m_arm_faults_flags &= ~ArmFaultsStatus::HARDWARE;
+  }
+  
+  // check for antenna faults
+  isFault = isFlagSet( J_ANT_PAN, msg->flags);
+  if (isFault) {
+    m_antenna_faults_flags |= PanTiltFaultsStatus::PAN_JOINT_LOCKED;
+  } else {
+    m_antenna_faults_flags &= ~PanTiltFaultsStatus::PAN_JOINT_LOCKED;
   }
 
-  setFaultsMessageHeader(arm_faults_msg);
-  m_arm_faults_msg_pub.publish(arm_faults_msg);
-  publishSystemFaultsMessage();
+  isFault = isFlagSet( J_ANT_TILT, msg->flags);
+  if (isFault) {
+    m_antenna_faults_flags |= PanTiltFaultsStatus::TILT_JOINT_LOCKED;
+  } else {
+    m_antenna_faults_flags &= ~PanTiltFaultsStatus::TILT_JOINT_LOCKED;
+  }
+
+  // update system faults
+  if (ArmFaultsStatus::NONE == m_arm_faults_flags) {
+    m_system_faults_flags &= ~SystemFaultsStatus::ARM_EXECUTION_ERROR;
+  } else {
+    m_system_faults_flags |= SystemFaultsStatus::ARM_EXECUTION_ERROR;
+  }
+  if (PanTiltFaultsStatus::NONE == m_antenna_faults_flags) {
+    m_system_faults_flags &= ~SystemFaultsStatus::PAN_TILT_EXECUTION_ERROR;
+  } else {
+    m_system_faults_flags |= SystemFaultsStatus::PAN_TILT_EXECUTION_ERROR;
+  }
+
+  // publish updated faults messages
+  owl_msgs::ArmFaultsStatus arm_faults_msg;
+  publishFaultsMessage(m_arm_faults_msg_pub, arm_faults_msg, m_arm_faults_flags);
```

Comment:
This comment applies to many similar instances in this file.  These two lines could be combined into just:

`publishFaultsMessage(m_arm_faults_msg_pub, owl_msgs::ArmFaultsStatus(), m_arm_faults_flags);`

In either case, passing a static (stack-allocated) object can be unsafe, but it's fine in this case because the object is no longer referenced after the called function returns.